File: library/dd-sdk-android-trace/src/main/java/com/datadog/trace/common/writer/Writer.java
/*
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2016-Present Datadog, Inc.
 */

package com.datadog.trace.common.writer;

import com.datadog.opentracing.DDSpan;
import java.io.Closeable;
import java.util.List;

/** A writer is responsible to send collected spans to some place */
public interface Writer extends Closeable {

  /**
   * Write a trace represented by the entire list of all the finished spans
   *
   * @param trace the list of spans to write
   */
  void write(List<DDSpan> trace);

  /** Start the writer */
  void start();

  /**
   * Indicates to the writer that no future writing will come and it should terminates all
   * connections and tasks
   */
  @Override
  void close();

  /** Count that a trace was captured for stats, but without reporting it. */
  void incrementTraceCount();

}

File: dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/Writer.kt
/*
 * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
 * This product includes software developed at Datadog (https://www.datadoghq.com/).
 * Copyright 2016-2019 Datadog, Inc.
 */

package com.datadog.android.core.internal.data

import com.datadog.android.log.internal.domain.Log

/**
 * Writes a log to a persistent location, for them to be sent at a later time (undefined).
 * @see [Reader]
 */
internal interface Writer {

    fun writeLog(log: Log)
}
